### PR TITLE
First cut at switching to new k8s deployer

### LINKED ISSUE #47: Move K8S deployer to spring-cloud-kubernetes-deployer
As a developer, I'd want to move implementation of `KubernetesModuleDeployer` SPI to `spring-cloud-kubernetes-deployer` project, so they can be consumed as library.

**Acceptance:**
- All the existing `stream` and `task` unit tests run successfully
- The project successfully builds 
- Docs updated

### PR DESCRIPTION
**Pull request by trisberg**
- Add AutoConfiguration and register DockerResourceLoader
- New Kafka kubernetes def
- Add kafka and rabbit applications definitions
- switch to rabbit using:
  
  --spring.cloud.dataflow.registry.populator.locations=classpath:META-INF/rabbit-applications.properties

Resolves spring-cloud/spring-cloud-dataflow-admin-kubernetes#47
Resolves spring-cloud/spring-cloud-dataflow-admin-kubernetes#59
Resolves spring-cloud/spring-cloud-dataflow-admin-kubernetes#30

### CODE REVIEW

**Comment by trisberg** [comment]
Depends on spring-cloud/spring-cloud-kubernetes-deployer#8

**Comment by trisberg** [comment]
Rebased

**Comment by jvalkeal** [comment]
I don't have environment to test but code looks good and is exactly what I've been doing on yarn side.

**Comment by jvalkeal** [comment]
I haven't really paid that much attention to those docker images, I've seen it using `latest` tag, but shouldn't those really be versioned? With snapshot's `latest` is probably equivalent to maven but what comes for the release, we should have a versioned docker image?

**Comment by trisberg** [comment]
Yes, once we release the first milestone of the app-starters we should tag them and use these tags for the servers. Until then we should use latest. @markpollack

**Comment by markpollack** [comment]
Yes, when we release it will be a tag such a 1.0.0.M1 etc.   There maybe something nice in the fabric8 mvn plugin for this, but I have't found it yet, e.g.   BUILD-SNAPSHOT -> :latest but 1.0.0.M1 -> :1.0.0.M1 ....

File: spring-cloud-dataflow-server-kubernetes/src/main/java/org/springframework/cloud/dataflow/server/kubernetes/KubernetesDataFlowServer.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.cloud.dataflow.server.EnableDataFlowServer;
+import org.springframework.context.annotation.ComponentScan;
```
Comment (markpollack) [inline spring-cloud-dataflow-server-kubernetes/src/main/java/org/springframework/cloud/dataflow/server/kubernetes/KubernetesDataFlowServer.java]: componentscan import not needed.

Comment (trisberg) [inline spring-cloud-dataflow-server-kubernetes/src/main/java/org/springframework/cloud/dataflow/server/kubernetes/KubernetesDataFlowServer.java]: fixed

Comment (markpollack) [inline spring-cloud-dataflow-server-kubernetes/src/main/java/org/springframework/cloud/dataflow/server/kubernetes/KubernetesDataFlowServer.java]: i fixed it on merge...

**Comment by markpollack** [comment]
when testing I had to bump the memory up... I picked 1024Mi....for time | log.  Maybe the deployer default  should be higher?

Also, getting this ZK error in both time and log.

for time it is 

```

11:29:15.000
2016-04-22 15:29:15.348 INFO 1 --- [nio-8080-exec-7] org.apache.zookeeper.ZooKeeper : Initiating client connection, connectString=10.159.247.86:2181 sessionTimeout=10000 watcher=org.I0Itec.zkclient.ZkClient@4bfdf6a4
11:29:15.000
2016-04-22 15:29:15.351 INFO 1 --- [159.247.86:2181] org.I0Itec.zkclient.ZkEventThread : Starting ZkClient event thread.
11:29:15.000
2016-04-22 15:29:15.352 INFO 1 --- [59.247.86:2181)] org.apache.zookeeper.ClientCnxn : Opening socket connection to server 10.159.247.86/10.159.247.86:2181. Will not attempt to authenticate using SASL (unknown error)
11:29:15.000
2016-04-22 15:29:15.353 INFO 1 --- [59.247.86:2181)] org.apache.zookeeper.ClientCnxn : Socket connection established to 10.159.247.86/10.159.247.86:2181, initiating session
11:29:15.000
2016-04-22 15:29:15.356 INFO 1 --- [59.247.86:2181)] org.apache.zookeeper.ClientCnxn : Session establishment complete on server 10.159.247.86/10.159.247.86:2181, sessionid = 0x1543e8520a4006d, negotiated timeout = 10000
11:29:15.000
2016-04-22 15:29:15.357 INFO 1 --- [c-7-EventThread] org.I0Itec.zkclient.ZkClient : zookeeper state changed (SyncConnected)
11:29:15.000
2016-04-22 15:29:15.360 INFO 1 --- [159.247.86:2181] org.I0Itec.zkclient.ZkEventThread : Terminate ZkClient event thread.
11:29:15.000
2016-04-22 15:29:15.363 INFO 1 --- [nio-8080-exec-7] org.apache.zookeeper.ZooKeeper : Session: 0x1543e8520a4006d closed
11:29:15.000
2016-04-22 15:29:15.364 INFO 1 --- [c-7-EventThread] org.apache.zookeeper.ClientCnxn : EventThread shut down
11:29:25.000
2016-04-22 15:29:25.345 INFO 1 --- [nio-8080-exec-6] org.apache.zookeeper.ZooKeeper : Initiating client connection, connectString=10.159.247.86:2181 sessionTimeout=10000 watcher=org.I0Itec.zkclient.ZkClient@3e3dbb06
11:29:25.000
2016-04-22 15:29:25.346 INFO 1 --- [159.247.86:2181] org.I0Itec.zkclient.ZkEventThread : Starting ZkClient event thread.
```

and for the sink it is

```
11:31:50.000
2016-04-22 15:31:50.878 INFO 1 --- [pool-1-thread-1] log.sink : 04/22/16 15:31:50
11:31:51.000
2016-04-22 15:31:51.879 INFO 1 --- [pool-1-thread-1] log.sink : 04/22/16 15:31:51
11:31:52.000
2016-04-22 15:31:52.879 INFO 1 --- [pool-1-thread-1] log.sink : 04/22/16 15:31:52
11:31:53.000
2016-04-22 15:31:53.879 INFO 1 --- [pool-1-thread-1] log.sink : 04/22/16 15:31:53
11:31:54.000
2016-04-22 15:31:54.171 INFO 1 --- [nio-8080-exec-6] org.apache.zookeeper.ZooKeeper : Initiating client connection, connectString=10.159.247.86:2181 sessionTimeout=10000 watcher=org.I0Itec.zkclient.ZkClient@5b6a6811
11:31:54.000
2016-04-22 15:31:54.173 INFO 1 --- [159.247.86:2181] org.I0Itec.zkclient.ZkEventThread : Starting ZkClient event thread.
11:31:54.000
2016-04-22 15:31:54.173 INFO 1 --- [59.247.86:2181)] org.apache.zookeeper.ClientCnxn : Opening socket connection to server 10.159.247.86/10.159.247.86:2181. Will not attempt to authenticate using SASL (unknown error)
11:31:54.000
2016-04-22 15:31:54.173 INFO 1 --- [59.247.86:2181)] org.apache.zookeeper.ClientCnxn : Socket connection established to 10.159.247.86/10.159.247.86:2181, initiating session
11:31:54.000
2016-04-22 15:31:54.176 INFO 1 --- [59.247.86:2181)] org.apache.zookeeper.ClientCnxn : Session establishment complete on server 10.159.247.86/10.159.247.86:2181, sessionid = 0x1543e8520a40096, negotiated timeout = 10000
11:31:54.000
2016-04-22 15:31:54.176 INFO 1 --- [c-6-EventThread] org.I0Itec.zkclient.ZkClient : zookeeper state changed (SyncConnected)
11:31:54.000
2016-04-22 15:31:54.189 INFO 1 --- [159.247.86:2181] org.I0Itec.zkclient.ZkEventThread : Terminate ZkClient event thread.
11:31:54.000
2016-04-22 15:31:54.192 INFO 1 --- [nio-8080-exec-6] org.apache.zookeeper.ZooKeeper : Session: 0x1543e8520a40096 closed
11:31:54.000
2016-04-22 15:31:54.192 INFO 1 --- [c-6-EventThread] org.apache.zookeeper.ClientCnxn : EventThread shut down
11:31:54.000
2016-04-22 15:31:54.880 INFO 1 --- [pool-1-thread-1] log.sink : 04/22/16 15:31:54
11:31:55.000
2016-04-22 15:31:55.881 INFO 1 --- [pool-1-thread-1] log.sink : 04/22/16 15:31:55
11:31:56.000
2016-04-22 15:31:56.881 INFO 1 --- [pool-1-thread-1] log.sink : 04/22/16 15:31:56
11:31:57.000
2016-04-22 15:31:57.881 INFO 1 --- [pool-1-thread-1] log.sink : 04/22/16 15:31:57
```

**Comment by markpollack** [comment]
squashed and merged as e8fb78483bc6dc2ae05887151dff6dff85797c6a

**Pull request closed without merging**